server/src/interfaces.ts

```diff
@@ -82,10 +82,21 @@ export type ListingStatus =
   | 'unsuccessful';
 
 /**
- * ListingPayload Interface that contains the fields of the payload that
- * would be sent to create a Listing Entity.
+ * ListingComputedProperties Interface that contains the fields of a Listing that
+ * are not provided by the client, but computed by the server.
+ */
+export interface ListingComputedProperties {
+  numCommits?: number;
+  numPaid?: number;
+  numCompleted?: number;
+  listingStatus?: ListingStatus;
+}
```

Comment:
Actually why did you make these fields optional?

Reply:
@hzxcaryn more for future use case. LIke for example, if a listing passes its deadline, then the `listingStatus` would be changed but not the rest. Then can use this interface as a parameter for that function.

Reply:
@karenfrilya97 Ohh I see I see.

Do you think leaving it like this (aka `ListingComputedProperties` being fully optional, and using `Required<ListingComputedProperties>` when you need all of the fields to be non-optional) is better or changing it to something like

```
export interface ListingComputedProperties {
  numCommits: number;
  numPaid: number;
  numCompleted: number;
  listingStatus: ListingStatus;
}

export type ListingComputedPropertiesPartial = Partial<ListingComputedProperties>;
```
and then using the `ListingComputedPropertiesPartial` when you want all the fields to be optional?
Personally I like this more because it feels like "these are all the properties, but I just wanna update some of them, hence I use the partial version", but it might just be my own preference and it's up to you to decide!

Reply:
@hzxcaryn Ohh, that makes sense. Will do that! Thanks for the suggestion